Core/Types/ModuleInstallDescriptor.cs

```diff
@@ -330,78 +330,48 @@ private bool IsWanted(string path, int? matchWhere)
         /// Throws a BadMetadataKraken if the stanza resulted in no files being returned.
         /// </summary>
         /// <exception cref="BadInstallLocationKraken">Thrown when the installation path is not valid according to the spec.</exception>
-        public List<InstallableFile> FindInstallableFiles(ZipFile zipfile, KSP ksp)
+        public List<InstallableFile> FindInstallableFiles(ZipFile zipfile, GameInstance ksp)
         {
             string installDir;
             var files = new List<InstallableFile>();
 
             // Normalize the path before doing everything else
-            string install_to = KSPPathUtils.NormalizePath(this.install_to);
+            string install_to = CKANPathUtils.NormalizePath(this.install_to);
 
-            if (install_to == "GameData" || install_to.StartsWith("GameData/"))
+            if (install_to == ksp.game.PrimaryModDirectoryRelative
```

Comment:
This made `ksp` no longer optional, it throws a `NullReferenceException` because we often pass `null` for `ksp` in Netkan. Same for the `else` branch below, there is a potential null reference in the `default` block.
I think we're fine jumping this whole if-else when `ksp == null` even if it means skipping a bunch of these sanity checks, we still invoke this other times with a non-null `ksp` (e.g. via `HasInstallableFiles()` in `InstallsFilesValidator`).

Reply:
Now creating `GameInstance`s to pass.

Reply:
Okay, makes sense. I still think `FindInstallableFiles()` should handle it gracefully, and the code is a bit messy as is right now. I did a PR to your branch which I think solves this problem nicely: https://github.com/HebaruSan/CKAN/pull/15